### PR TITLE
feat(eslint-plugin): [member-ordering] support static blocks

### LINKED ISSUE #5404: Bug: [member ordering] Static initialization blocks are regarded as public instance methods.
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=4.7.2&sourceType=module&code=MYGwhgzhAEAqCmEAu0DeAoa1liQS2GiQE8AHeLAXmgHIAZGgbk2mAHsA7ZAJwFdgkbbgAoAlGmgBfdNPShIMBMgBMaFjnyFUUluy5I+AoWLTTJQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AW0XYCNFpaA9tAAmfZgHNIYDAG0ofaEMgAacNghZ12SKIBmAQwT4psqLsqJ4wlVGSVxTffljRENyGQFNk+OGXxKylCc+AAWAtZgALqqWtpk8PrIyCgmYDJqcTjEVABuToi0Pk6UZLTmltaxWdmw3Ixlxfil5RZWKplZkDmU+fiFTS0h4VWdcd11DUX4JWXDER012XkFtMzFTGSFFe3VNRP1Leszm9tto0tQnt6+sP6BY1rdK-1rN-qntPMXSwdTxx8tl9EGEFo8IFFHgBfTqQiAwqFAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
class Test {
  static type  = 'L';
  constructor() { }
}

class Test2 {
  static { }
  constructor() {}
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/member-ordering" : [ "error",
      {
        "default" : [ "field", "signature", "constructor", "method" ],
        "classes" : [
          "private-static-field",
          "public-static-field",
          "private-static-method",
          "public-static-method",
          "private-instance-field",
          "public-instance-field",
          "constructor",
          "private-instance-method",
          "public-instance-method"
        ]
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No error.

### Actual Result

```
Member constructor should be declared before all public instance method definitions. 8:3 - 8:19

```

### Additional Info

_No response_

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.31.0` |
| `@typescript-eslint/parser`            | `5.31.0` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.20.0` |
| `node`                                 | `18.7.0` |

### PR DESCRIPTION
**Pull request by Josh-Cena**
## PR Checklist

-   [x] Addresses an existing open issue: fixes #5404
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

### CODE REVIEW

File: packages/eslint-plugin/docs/rules/member-ordering.md [hunk 1]
```diff
@@ -94,6 +94,9 @@ The default configuration looks as follows:
 
     "field",
 
+    // Static initialization
+    "static-iniatialization",
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/docs/rules/member-ordering.md:98]: Lil typo:

```suggestion
    "static-initialization",
```

File: packages/eslint-plugin/tests/rules/member-ordering.test.ts [hunk 2]
```diff
@@ -3979,6 +3999,57 @@ class Foo {
...
+          column: 3,
+        },
+      ],
+    },
+    {
+      code: `
+class Foo {
+  m() {}
+  f = 1;
+  static {}
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/tests/rules/member-ordering.test.ts:4037]: Could you also add test cases for `static {}` being intermixed with other blocks? Just to make sure there are no funky edge cases in ordering behavior.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Looks great, thanks for sending this in so quickly!

Just requesting changes on a bit more test coverage.

File: packages/eslint-plugin/tests/rules/member-ordering.test.ts (hunk 2 above)
Comment (Josh-Cena) [reply packages/eslint-plugin/tests/rules/member-ordering.test.ts:4047]: By continuity principle (something I just made up) the tests should be sufficient. Let me know if there are specific quirks worth testing that you have in mind.

**Review by JoshuaKGoldberg — APPROVED** [review]
Perfect 😄 thanks!

**Pull request merged**